src/zenml/config/docker_settings.py

```diff
@@ -192,8 +209,10 @@ class DockerSettings(BaseSettings):
     )
     python_package_installer_args: Dict[str, Any] = {}
     replicate_local_python_environment: Optional[
-        Union[List[str], PythonEnvironmentExportMethod]
+        Union[List[str], PythonEnvironmentExportMethod, bool]
     ] = Field(default=None, union_mode="left_to_right")
+    pyproject_path: Optional[str] = None
+    pyproject_export_command: Optional[List[str]] = None
```

Comment:
As far as I can see, it is possible to set this as follows: `DockerSettings(pyproject_export_command=[......])` without an explicit `pyproject_path` as long as there is a `pyproject.toml` file to work with, in the current source root. But as soon as someone puts an integration, requirement or the replicate flag, this becomes unused.

Reply:
Perhaps a validator can be helpful to detect this early.

Reply:
What would you validate here? I don't see the problem of this being unused if there is no `pyproject.toml` to be honest

Reply:
I would validate the following two scenarios:
- If `pyproject_export_command` is set, there needs to be a `pyproject.toml` in the directory.
- If `required_integrations` are set, `pyproject_export_command ` is unused. 

However, both are not so critical, I just thought early detection might help us improve the UX a bit.